services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/AttachmentEntity.kt

```diff
@@ -184,13 +185,21 @@ interface HankeAttachmentRepository : JpaRepository<HankeAttachmentEntity, UUID>
     @Modifying
     @Query("DELETE FROM HankeAttachmentEntity WHERE id = :id")
     fun deleteAttachment(id: UUID)
+
+    @Query("SELECT count(h.id) FROM HankeAttachmentEntity h WHERE h.hanke.id = :hankeId")
+    fun countByHanke(@Param("hankeId") hankeId: Int): Int
 }
 
 @Repository
 interface ApplicationAttachmentRepository : JpaRepository<ApplicationAttachmentEntity, UUID> {
     @Modifying
     @Query("DELETE FROM ApplicationAttachmentEntity WHERE id = :id")
     fun deleteAttachment(id: UUID)
+
+    @Query(
+        "SELECT count(a.id) FROM ApplicationAttachmentEntity a WHERE a.application.id = :applicationId"
+    )
+    fun countByApplication(@Param("applicationId") applicationId: Long): Int
```

Comment:
Can't this be done with a derived query?
```suggestion
    fun countByApplicationId(applicationId: Long): Long
```

Reply:
countByApplicationId uses count(*) and using count(id) may have some performance benefits as it does not read the whole row. But this just may be a case of over engineering so I'll use countByApplicationId(applicationId: Long) for the sake of simplicity.

Reply:
Changed as suggested.

Reply:
`count(*)` doesn't read the whole row, it only needs to find out how many there are. `count(id)` counts how many rows there are where id is not null, so it's generally slower.

Of course, with a limit of 20 attachments per application, this is all very theoretical. Unless there's a measurable benefit for doing otherwise, the choice should always be in favour of simplicity and clarity.